File: app/src/main/java/kh/edu/rupp/ite/mad_project/Adapter/FavoriteAdapter.java
package kh.edu.rupp.ite.mad_project.Adapter;

import android.annotation.SuppressLint;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.squareup.picasso.Picasso;

import java.util.List;

import kh.edu.rupp.ite.mad_project.R;
import kh.edu.rupp.ite.mad_project.model.HomeProducts;

public class FavoriteAdapter extends RecyclerView.Adapter<FavoriteAdapter.FavoriteViewHolder> {

    private final List<HomeProducts> favoriteProducts;

    public FavoriteAdapter(List<HomeProducts> favoriteProducts) {
        this.favoriteProducts = favoriteProducts;
    }

    @NonNull
    @Override
    public FavoriteViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.view_holder_item, parent, false);
        return new FavoriteViewHolder(view);
    }

    @SuppressLint("SetTextI18n")
    @Override
    public void onBindViewHolder(@NonNull FavoriteViewHolder holder, int position) {
        HomeProducts product = favoriteProducts.get(position);
        holder.titleTextView.setText(product.getTitle());
        holder.priceTextView.setText("$" + product.getPrice());

        String imageUrl = product.getImage();
        if (imageUrl != null && !imageUrl.isEmpty()) {
            Picasso.get().load(imageUrl).into(holder.productImageView);
        } else {
            holder.productImageView.setImageResource(R.drawable.computer); // Placeholder image
        }

        // Set click listener for product item if needed
        holder.itemView.setOnClickListener(v -> {
            // Handle item click, for example, navigate to product details
        });
    }

    @Override
    public int getItemCount() {
        return favoriteProducts.size();
    }

    public static class FavoriteViewHolder extends RecyclerView.ViewHolder {
        TextView titleTextView, priceTextView;
        ImageView productImageView;

        public FavoriteViewHolder(@NonNull View itemView) {
            super(itemView);
            titleTextView = itemView.findViewById(R.id.productTitle);
            priceTextView = itemView.findViewById(R.id.productPrice);
            productImageView = itemView.findViewById(R.id.product_image);
        }
    }
}

File: app/src/main/java/kh/edu/rupp/ite/mad_project/Adapter/FavoriteAdapter.kt
package kh.edu.rupp.ite.mad_project.Adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import kh.edu.rupp.ite.mad_project.R
import kh.edu.rupp.ite.mad_project.model.FavoriteProductResponse

class FavoriteProductAdapter(private var favoriteProducts: MutableList<FavoriteProductResponse>) : RecyclerView.Adapter<FavoriteProductAdapter.FavoriteViewHolder>() {

    class FavoriteViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val productName: TextView = view.findViewById(R.id.tvProductName)
        val productPrice: TextView = view.findViewById(R.id.tvProductPrice)
        val imgProduct: ImageView = view.findViewById(R.id.imgProduct)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FavoriteViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.view_holder_favorite, parent, false)
        return FavoriteViewHolder(view)
    }

    override fun onBindViewHolder(holder: FavoriteViewHolder, position: Int) {
        val favoriteProduct = favoriteProducts[position]
        holder.productName.text = favoriteProduct.product.name
        holder.productPrice.text = favoriteProduct.product.price

        // Load the image using Glide
        Glide.with(holder.itemView.context)
            .load(favoriteProduct.product.image)
            .into(holder.imgProduct)
    }

    override fun getItemCount() = favoriteProducts.size

    fun updateData(newFavoriteProducts: List<FavoriteProductResponse>) {
        favoriteProducts.clear()
        favoriteProducts.addAll(newFavoriteProducts)
        notifyDataSetChanged()
    }
}
